layers/ip6.go

```diff
@@ -280,165 +229,191 @@ func decodeIPv6(data []byte, p gopacket.PacketBuilder) error {
 	return p.NextDecoder(ip6.NextHeader)
 }
 
-func (i *IPv6HopByHop) SerializeTo(b gopacket.SerializeBuffer, opts gopacket.SerializeOptions) error {
-	var bytes []byte
-	var err error
+type IPv6HeaderTLVOption interface {
```

Comment:
Originally, ipv6headertlvoption was not exported... is there a reason to export it here?

If it IS being exported, we'll need to add a comment for the interface and also for its functions.

I'm worried about the use of an interface here, since interfaces generally tend to require additional mallocs.  If an interface survives outside a function call, like in decodeIPv6HeaderTLVOption, a malloc will happen.  It looks like the old decodeIPv6HeaderTLVOption also malloc'd, but I'm not sure it would have had to.